### PR TITLE
Allow the use of checks and assignments in initial block

### PR DESCRIPTION
**Pull request by LendvaiDenes**

### CODE REVIEW

File: plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/AbstractContractStatechartGeneration.xtend [hunk 1]
```diff
@@ -466,13 +467,48 @@ abstract class AbstractContractStatechartGeneration {
...
+	def protected retargetAllEventParamRefs(EObject container, boolean reversed) {
+		val eventParamRefs = ecoreUtil.getAllContentsOfType(container, EventParameterReferenceExpression)
 		for (eventParamRef : eventParamRefs) {
 			eventParamRef.port = getPort(getNameOfNewPort(eventParamRef.port, reversed))
 			eventParamRef.event = getEvent(eventParamRef.event.name, eventParamRef.port)
 		}
+	}
+
+	def protected addAssignmentsToTransition(Iterable<ScenarioAssignmentStatement> assignments, Transition transition) {
+		for (assignment : assignments) {
```
Comment (grbeni) [inline plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/AbstractContractStatechartGeneration.xtend:479]: Why not a single line: `transition.effects += assignments.clone`?

File: plugins/scenario/hu.bme.mit.gamma.scenario.model/src/hu/bme/mit/gamma/scenario/model/reduction/SimpleScenarioGenerator.java [hunk 2]
```diff
@@ -143,8 +143,8 @@ private InitialBlock handleInitBlockCopy() {
 			return null;
 		}
 		InitialBlock initBloc = factory.createInitialBlock();
-		for (ModalInteraction modalInteraction : base.getInitialblock().getModalInteractions()) {
-			initBloc.getModalInteractions().add((ModalInteraction) doSwitch(modalInteraction));
+		for (InteractionDefinition interaction : base.getInitialblock().getInteractions()) {
+			initBloc.getInteractions().add((InteractionDefinition) doSwitch(interaction));
```
Comment (grbeni) [inline plugins/scenario/hu.bme.mit.gamma.scenario.model/src/hu/bme/mit/gamma/scenario/model/reduction/SimpleScenarioGenerator.java:147]: Is the `(InteractionDefinition)` casting right in front of `doSwitch(interaction)` necessary?

Comment (LendvaiDenes) [reply plugins/scenario/hu.bme.mit.gamma.scenario.model/src/hu/bme/mit/gamma/scenario/model/reduction/SimpleScenarioGenerator.java:147]: Yes, as the doSwitch function returns an EObject

File: plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/AbstractContractStatechartGeneration.xtend (hunk 1 above)
Comment (LendvaiDenes) [reply plugins/scenario/hu.bme.mit.gamma.scenario.statechart.generator/src/hu/bme/mit/gamma/scenario/statechart/generator/AbstractContractStatechartGeneration.xtend:479]: This is how I tired it first, but I received a parsing exception

**Review by grbeni — APPROVED** [review]
Nice, thank you.

**Pull request merged**